lib/logstash/filters/aggregate.rb

```diff
@@ -384,10 +433,31 @@ def remove_expired_maps()
     min_timestamp = Time.now - @timeout
     
     @@mutex.synchronize do
+
       @@aggregate_maps.delete_if do |key, element| 
         if (element.creation_timestamp < min_timestamp)
-          if (@push_previous_map_as_event)
-            events_to_flush << LogStash::Event.new(element.map)
+          if (@push_previous_map_as_event) || (@push_map_as_event_on_timeout)
+            event = LogStash::Event.new(element.map)        
+            
+            if @timeout_task_id_field
+              event_to_yield[@timeout_task_id_field] = task_id
+            end
+
+            # Call code block if available
+            if @timeout_code
+              begin
+                @timeout_codeblock.call(event)
+                events_to_flush << event
+                noError = true
+              rescue => exception
+                @logger.error("Aggregate exception occurred. Error: #{exception} ; Code: #{@timeout_code} ; EventData: #{event.instance_variable_get('@data')}")
```

Comment:
Added this. There are also a bunch of non existent variables (i blame my editor :P) . E.g. i create "event" but I add the task_id_field to "event_to_yield". I changed all of them to be "event_to_yield" for consistency